File: src/consumer/EventTriangleAPI.Consumer.BusinessLogic/Models/TransactionDto.cs
using EventTriangleAPI.Shared.DTO.Enums;

namespace EventTriangleAPI.Consumer.BusinessLogic.Models;

public record TransactionDto(
    Guid Id, 
    string FromUserId,
    string ToUserId,
    decimal Amount,
    TransactionState TransactionState,
    TransactionType TransactionType,
    DateTime CreatedAt);

File: src/authorization/EventTriangle.Client/src/app/types/models/consumer/TransactionDto.ts
import {TransactionState} from "../../enums/TransactionState";
import {TransactionType} from "../../enums/TransactionType";

export interface TransactionDto {
  id: string;
  fromUserId: string;
  toUserId: string;
  amount: number;
  transactionState: TransactionState;
  transactionType: TransactionType;
  createdAt: string;
}
